src/UtilMsg.chpl

```diff
@@ -165,71 +121,52 @@ module UtilMsg {
 
     Implements the 'constant' mode of numpy.pad: https://numpy.org/doc/stable/reference/generated/numpy.pad.html
   */
-  @arkouda.registerND
-  proc padMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, param nd: int): MsgTuple throws {
-    param pn = Reflection.getRoutineName();
-
-    const name = msgArgs.getValueOf("name"),
-          padWidthBefore = msgArgs.get("padWidthBefore").getTuple(nd),
-          padWidthAfter = msgArgs.get("padWidthAfter").getTuple(nd),
-          padValsBefore = msgArgs.get("padValsBefore"),
-          padValsAfter = msgArgs.get("padValsAfter"),
-          rname = st.nextName();
-
-    var gEnt: borrowed GenSymEntry = getGenericTypedArrayEntry(name, st);
-
-    proc doPad(type t): MsgTuple throws {
-      const e = toSymEntry(gEnt, t, nd);
-
-      const pvb = padValsBefore.toScalarArray(t, nd),
-            pva = padValsAfter.toScalarArray(t, nd);
-
-      // compute the padded shape
-      var outShape: nd*int;
-      for i in 0..<nd do outShape[i] = padWidthBefore[i] + e.tupShape[i] + padWidthAfter[i];
-
-      var p = st.addEntry(rname, (...outShape), t);
-
-      // copy the original array into the padded array
-      const dOffset = e.a.domain.translate(padWidthBefore);
-      p.a[dOffset] = e.a;
-
-      // starting with the last dimension, pad the array (i.e., dimension 0 overwrites dimension 1 in the corners, etc.)
-      for rank in 0..<nd {
-        var beforeSlice, afterSlice: nd*range;
-        for i in 0..<nd {
-          // TODO: compute the exact slice for each pad-section so these assignments can be done
-          // in parallel and to avoid accessing the corners of the array unnecessarily (which
-          // could result in additional comm for large pad widths)
-          if i == rank {
-            beforeSlice[i] = 0..<padWidthBefore[i];
-            afterSlice[i] = (outShape[i]-padWidthAfter[i])..<outShape[i];
-          } else {
-            beforeSlice[i] = 0..<outShape[i];
-            afterSlice[i] = 0..<outShape[i];
-          }
+  @arkouda.instantiateAndRegister
+  proc pad(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, type array_dtype, param array_nd: int): MsgTuple throws 
+    where (array_dtype == int) || (array_dtype == uint(8)) || (array_dtype == uint(64)) || (array_dtype == real) || (array_dtype == bool) {
+
+    const e = st[msgArgs["name"]]: SymEntry(array_dtype, array_nd),
+          padWidthBefore = msgArgs["padWidthBefore"].getTuple(array_nd),
+          padWidthAfter = msgArgs["padWidthAfter"].getTuple(array_nd),
```

Comment:
Can you use `toScalarTuple(int, array_nd)` here? The plan is to get rid of `getTuple` eventually.

Reply:
I fixed this.  Thanks for pointing it out.